client/resource_group/controller/controller.go

```diff
@@ -261,11 +267,26 @@ func (c *ResourceGroupsController) cleanUpResourceGroup(ctx context.Context) err
 	for _, group := range groups {
 		latestGroups[group.GetName()] = struct{}{}
 	}
-	// TODO: maybe we should also clean up those resource groups that have not been used for a long time.
 	c.groupsController.Range(func(key, value any) bool {
 		resourceGroupName := key.(string)
 		if _, ok := latestGroups[resourceGroupName]; !ok {
 			c.groupsController.Delete(key)
+			return true
+		}
+
+		gc := value.(*groupCostController)
+		// Check for stale resource groups, which will be deleted when consumption is continuously unchanged.
+		gc.mu.Lock()
+		if equalRU(gc.mu.consumption, gc.run.consumption) {
+			gc.mu.Unlock()
+			if gc.tombstone {
+				c.groupsController.Delete(resourceGroupName)
+				return true
+			}
+			gc.tombstone = true
+		} else {
+			gc.mu.Unlock()
+			gc.tombstone = false
```

Comment:
```suggestion
		gc.mu.Lock()
		latestConsumption := gc.mu.consumption
		gc.mu.Unlock()
		if equalRU(latestConsumption, gc.run.consumption) {
			if gc.tombstone {
				c.groupsController.Delete(resourceGroupName)
				return true
			}
			gc.tombstone = true
		} else {
			gc.tombstone = false
```

Reply:
Done, but it's a pointer so needs to take a value to prevent data race